### PR TITLE
Fix failure_reason label for nexus_task_execution_failed metric on task timeout

### PR DESCRIPTION
**Pull request by bergundy**
We mistakenly reported timeouts as `internal_sdk_error`. I'll update the docs site as well.

### CODE REVIEW

File: test/nexus_test.go [hunk 1]
```diff
@@ -350,6 +353,20 @@ func TestNexusSyncOperation(t *testing.T) {
...
+		_, err := nexus.ExecuteOperation(ctx, nc, syncOp, "timeout", nexus.ExecuteOperationOptions{
+			// Force shorter timeout to speed up the test and get a response back.
+			Header: nexus.Header{nexus.HeaderRequestTimeout: "300ms"},
+		})
+		var unexpectedResponseErr *nexus.UnexpectedResponseError
+		require.ErrorAs(t, err, &unexpectedResponseErr)
+		require.Equal(t, nexus.StatusDownstreamTimeout, unexpectedResponseErr.Response.StatusCode)
+
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
+			tc.requireFailureCounter(t, service.Name, syncOp.Name(), "timeout")
```
Comment (Quinn-With-Two-Ns) [inline test/nexus_test.go:367]: Why only assert the failure counter? shouldn't `NexusTaskScheduleToStartLatency` be published?

Comment (bergundy) [reply test/nexus_test.go:367]: I figured it's covered in the other tests.

Comment (Quinn-With-Two-Ns) [reply test/nexus_test.go:367]: hm I think this case is different because we don't report E2E latency on timeout

Comment (bergundy) [reply test/nexus_test.go:370]: I know, I meant the other two metrics. In any case, I've added the tests.

**Review by Quinn-With-Two-Ns — APPROVED** [review]

**Comment by bergundy** [comment]
We mistakenly reported timeouts as `internal_sdk_error`.

**Pull request merged**